include/aspect/particle/world.h

```diff
@@ -21,729 +21,310 @@
-         *
-         * @param [in,out] particle The particle to find the cell for. This
-         * particle will be updated to indicate whether it is in the local
-         * subdomain or not.
-         * @param [in] cur_cell The current cell (level and index) being
-         * checked.
-         * @return The level and index of the active cell the particle is in.
-         * If no cell was found to contain the particle, return the
-         * level/index (-1, -1)
-         */
-        LevelInd find_cell(T &particle, const LevelInd &cur_cell)
-        {
-          typename parallel::distributed::Triangulation<dim>::cell_iterator         it, found_cell;
-          typename parallel::distributed::Triangulation<dim>::active_cell_iterator  ait;
-          LevelInd    res;
-
-          // First check the last recorded cell since particles will generally stay in the same area
-          if (!triangulation_changed)
-            {
-              found_cell = typename parallel::distributed::Triangulation<dim>::cell_iterator(triangulation, cur_cell.first, cur_cell.second);
-              if (found_cell != triangulation->end() && found_cell->point_inside(particle.get_location()) && found_cell->active())
-                {
-                  // If the cell is active, we're at the finest level of refinement and can finish
-                  particle.set_local(found_cell->is_locally_owned());
-                  return cur_cell;
-                }
-            }
-
-          // Check all the cells on level 0 and recurse down
-          for (it=triangulation->begin(0); it!=triangulation->end(0); ++it)
-            {
-              res = recursive_find_cell(particle, std::make_pair(it->level(), it->index()));
-              if (res.first != -1 && res.second != -1) return res;
-            }
-
-          // If we couldn't find it there, we need to check the active cells
-          // since the particle could be on a curved boundary not included in the
-          // coarse grid
-          for (ait=triangulation->begin_active(); ait!=triangulation->end(); ++ait)
-            {
-              if (ait->point_inside(particle.get_location()))
-                {
-                  particle.set_local(ait->is_locally_owned());
-                  return std::make_pair(ait->level(), ait->index());
-                }
-            }
-
-          // If it failed all these tests, the particle is outside the mesh
-          particle.set_local(false);
-          return std::make_pair(-1, -1);
-        };
+         * Calculate the cells containing each particle for all particles. If
+         * particles moved out of the domain of this process they will be send
+         * to their new process and inserted there. After this function call
+         * every particle is either on its current process and in its current
+         * cell, or deleted (if it could not find its new process or cell).
+         */
+        void
+        find_all_cells();
```

Comment:
The name and the documentation of the function are slightly at odds. The name of the function doesn't suggest that particles may actually be communicated to other processors. Maybe `associated_particles_with_local_or_remote_cells()`? Or maybe something shorter?